tests/testthat/test-render.R

```diff
@@ -276,3 +276,47 @@ test_that("render_website() works - clean render for each language", {
   expect_true(fs::file_exists(file.path(parent_dir, project_dir, "_site", "fr", "subdir", "about.html")))
   expect_false(fs::dir_exists(file.path(parent_dir, project_dir, "_site", "es", "subdir")))
 })
+
+test_that("render_website() fails when missing sidebar and languagelinks is set to sidebar", {
+  withr::local_envvar("BABELQUARTO_TESTS_URL" = "true")
+
+  parent_dir <- withr::local_tempdir()
+  project_dir <- "blop"
+  quarto_multilingual_website(
+    parent_dir = parent_dir,
+    project_dir = project_dir,
+    further_languages = c("es", "fr"),
+    main_language = "en"
+  )
+
+  config_path <- file.path(parent_dir, project_dir, "_quarto.yml")
+  config_lines <- brio::read_lines(config_path)
+  where_languagelinks <- which(grepl("  languagelinks:", config_lines))
+  config_lines[where_languagelinks] <- "  languagelinks: sidebar"
+  brio::write_lines(config_lines, config_path)
+
+  expect_error(withr::with_dir(parent_dir, render_website(project_dir)))
```

Comment:
this could be an `expect_snapshot(..., error = TRUE)`, or you could test for part of the error message using the `regexp` argument of `expect_error()` to make it more specific?

Reply:
Thanks for the suggestion. It's great to learn new testhat functions!

I tried the snapshot idea, but it seems to have some issues depending on the platform. I'll fix it later tonight or tomorrow by using a regexp to look for the error message instead. It should be less _fragile_.

Reply:
I solved it with the regexp argument. They are now specific to the error thrown by the checks for misconfiguration.